src/wayland-eglsurface.c

```diff
@@ -191,8 +195,19 @@ wlEglSendDamageEvent(WlEglSurface *surface, struct wl_event_queue *queue)
                           surface->dy);
     }
 
-    wl_surface_damage(surface->wlSurface, 0, 0,
-                      surface->width, surface->height);
+    if (n_rects == 0) {
+        wl_surface_damage(surface->wlSurface, 0, 0,
+                          surface->width, surface->height);
+    } else {
+        for (i = 0; i < n_rects; i++) {
+            wl_surface_damage_buffer(surface->wlSurface,
+                                     rects[i],
+                                     surface->height - rects[i+1],
+                                     rects[i+2],
+                                     rects[i+3]);
+        }
```

Comment:
You can only do this `if (wl_proxy_get_version(surface->wlSurface) >= WL_SURFACE_DAMAGE_BUFFER_SINCE_VERSION)`. If it's older then you have to fall back to damaging the full surface - and as said with maximum uint32 range rather than surface dimensions.

Reply:
Ah, that's how to do it. Thanks!